### PR TITLE
(#1384014) add remote-cryptsetup.target and remote-cryptsetup-pre.target 

### PR DESCRIPTION
**Pull request by lnykryn**

### CODE REVIEW

**Comment by jsynacek** [comment]
There's never enough reviews, so... LGTM.

**Review by msekletar — APPROVED** [review]
LGTM

File: src/cryptsetup/cryptsetup-generator.c [hunk 1]
```diff
@@ -196,10 +199,17 @@ static int create_disk(
...
+                                to = strjoin(arg_dest, "/cryptsetup.target.requires/", n, NULL);
+                        else
+                                to = strjoin(arg_dest, "/cryptsetup.target.wants/", n, NULL);
+                } else {
+                        if (!nofail)
+                                to = strjoin(arg_dest, "/remote-cryptsetup.target.requires/", n, NULL);
+                        else
+                                to = strjoin(arg_dest, "/remote-cryptsetup.target.wants/", n, NULL);
+                }
                 if (!to)
```
Comment (msekletar) [inline src/cryptsetup/cryptsetup-generator.c:213]: We leak memory pointed by ```to``` when ```symlink``` fails. This seems to be an issue in upstream as well.

Comment (msekletar) [reply src/cryptsetup/cryptsetup-generator.c:213]: However, I don't see this as a blocker. So overall, LGTM.

**Review by msekletar — APPROVED** [review]

File: src/cryptsetup/cryptsetup-generator.c (hunk 1 above)
Comment (lnykryn) [reply src/cryptsetup/cryptsetup-generator.c:213]: Fixed. Upstream uses a different code here.

**Pull request merged**